### ISSUE TITLE
Dashboard time range picker throws "Minified React error"

### ISSUE THREAD

**Issue by stevenvachon** [grafana/grafana#25749]
**What happened**:
This is thrown on each key stroke:
```
react-dom.production.min.js:52 Uncaught Error: Minified React error #188; visit https://reactjs.org/docs/error-decoder.html?invariant=188 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at react-dom.production.min.js:52
    at ot (react-dom.production.min.js:53)
    at Object.findDOMNode (react-dom.production.min.js:286)
    at n.onOutsideClick (ClickOutsideWrapper.tsx:43)
```

**What you expected to happen**:
No error thrown.

**How to reproduce it (as minimally and precisely as possible)**:
Open a dashboard's absolute time range form and type into the "from" or "to" fields. It doesn't matter whether there're any panels or not.

**Anything else we need to know?**:
This error causes problems with E2E tests.

**Environment**:
- Grafana version: master branch (7.1)
- Data source type & version: any data source, I think
- OS Grafana is installed on: nightly-master.plugins.grafana.net
- User OS & Browser: macOS 10.15.5, Chrome 83
- Grafana plugins:
- Others:

**Comment by ryantxu** [comment]
I am unable to reproduce... am I typing in the right place?

![image](https://user-images.githubusercontent.com/705951/85343800-b43fc080-b4a2-11ea-8d47-b2422712bd59.png)

![image](https://user-images.githubusercontent.com/705951/85343829-d20d2580-b4a2-11ea-9373-e4b48e395cb6.png)

**Comment by stevenvachon** [comment]
That's the right place. Did you type or use the time/calendar selector? If so, does the data update or are there errors in the console?

**Comment by ryantxu** [comment]
Ah ok -- the results execute OK, but there is an error in the console:

![image](https://user-images.githubusercontent.com/705951/85446020-dd1a9100-b548-11ea-9940-707e71c79523.png)

**Comment by torkelo** [comment]
I also just get that error in the console, I managed to get a crash once, but not the same "Minified React error"

**Comment by torkelo** [comment]
I have not been able to replicate the exact same error "Minified React error", but I do get the error ryan reported, and I have been able to get one crash occasionally, but not sure exactly what the replication steps are 

![Screenshot from 2020-06-25 16-07-10](https://user-images.githubusercontent.com/10999/85735576-0f212400-b6fe-11ea-961e-1f232ef8123b.png)

**Comment by mckn** [comment]
I think I know what it is. It has to do with the ClickOutsideWrapper and event listeners on elements being detached. I will open a PR to resolve this. Thanks for reporting it! 🙌

**Comment by mckn** [comment]
@stevenvachon this will hopefully unblock you: https://github.com/grafana/grafana/pull/25844. Will see if I can prevent the errors that @torkelo managed to trigger in a separate PR.

**Comment by mckn** [comment]
> I also just get that error in the console, I managed to get a crash once, but not the same "Minified React error"

I'm guessing now - but it is probably because we are running it in "debug" bundle and I guess the e2e are running in a release bundle.

**Comment by stevenvachon** [comment]
Yup, I should've built it and ran it locally in dev mode to save the time of having to expand the error.
